autils/archive/archive.py

```
@@ -0,0 +1,539 @@
+            # This check might be slightly redundant now but kept for safety.
+            if self.compression_type not in ("gz", "zst"):
+                raise ArchiveException(
+                    f"Unsupported standalone compression type for direct extraction: {self.compression_type}"
+                )
+
+        # Handle regular archives (zip and tar)
+        if self.is_tar:
+            self._engine.extraction_filter = getattr(
+                tarfile, "fully_trusted_filter", (lambda member, path: member)
+            )
+        self._engine.extractall(path)
```

Ruff (0.16.0)

[error] Uses of `tarfile.extractall()`

(S202)